### PR TITLE
internal/legacy: check authorization in gated charms

### PR DESCRIPTION
**Pull request by rogpeppe**
This closes a security loophole.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/920/
Test PASSed.

File: internal/legacy/api_test.go [hunk 1]
```diff
@@ -136,7 +136,7 @@ func (s *APISuite) TestCharmInfoNotFound(c *gc.C) {
 	})
 }
 
-func (s *APISuite) TestServerCharmInfo(c *gc.C) {
+func (s *APISuite) TestServeCharmInfo(c *gc.C) {
```
Comment (mhilton) [inline internal/legacy/api_test.go:139]: Good spot

**Comment by mhilton** [comment]
:+1: LGTM

**Comment by frankban** [comment]
:+1:

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://localhost:8080/job/charmstore-merge

**Pull request merged**